docs/sync/bitbucket_pipelines.md

```diff
@@ -0,0 +1,316 @@
+---
+title: Bitbucket Pipelines Integration
+category: 62cdf6722c2c1602a4b69643
+hidden: false
+---
+# Bitbucket Pipelines Integration
+
+## Quickstart
+
+Add the following "Phylum Analyze" step to a `bitbucket-pipelines.yml` configuration, in one or more pipeline start
+conditions:
+
+```yaml
+# Ensure these variables are set at the repository or workspace level:
+#  - `PHYLUM_API_KEY`: Phylum authentication token
+#  - `BITBUCKET_TOKEN`: repository, project, or workspace access token with `pullrequest` scope
+
+    - step:
+        name: Phylum Analyze
+        image: phylumio/phylum-ci:latest
+        clone:
+          depth: full
+        script:
+          - phylum-ci
+```
+
+## Overview
+
+Once configured for a repository, the Bitbucket Pipelines integration will provide analysis of project
+dependencies from a lockfile. This can happen in a branch or default pipeline as a result of a commit or in a pull
+request (PR) pipeline. For PR pipelines, analyzed dependencies will include any that are added/modified in the PR.
+For branch pipelines, the analyzed dependencies will be determined by comparing the lockfile in the branch to the
+default branch. **All** dependencies will be analyzed when the branch pipeline is run on the default branch.
+
+The results will be provided in the pipeline logs and provided as a comment on the PR. The CI job will return an
+error (i.e., fail the build) if any of the analyzed dependencies fail to meet the project risk thresholds for any
+of the five Phylum risk domains:
+
+* Vulnerability (aka `vul`)
+* Malicious Code (aka `mal`)
+* Engineering (aka `eng`)
+* License (aka `lic`)
+* Author (aka `aut`)
+
+See [Phylum Risk Domains][risk_domains] documentation for more detail.
+
+**NOTE**: It is not enough to have the total project threshold set. Individual risk domain threshold values must be
+set, either in the UI or with `phylum-ci` options, in order to enable analysis results for CI. Otherwise, the risk
+domain is considered disabled and the threshold value used will be zero (0).
+
+There will be no comment if no dependencies were added or modified for a given PR.
```

Comment:
I'm sure this applies to other ecosystems too, but it seems like an interesting decision.

Why do we not post any comment when no dependencies were added? I'd assume the idea is just to not clutter the output, but I feel like at that point the same could be said about success with new dependencies?

Reply:
Correct...the decision was made long ago to reduce the amount of PR comments made by Phylum. The thinking was that if a PR did not add a new dependency (which could include a new version of an _existing_ dependency), then there should not be any mention by Phylum. However, if a lockfile was modified to include a new dependency, there should be _some_ mention of it...even if it was a successful analysis...just so that it is clear that the lockfile _was_ analyzed.